### DISCUSSION TITLE
How to return custom statistics

### DISCUSSION THREAD

**Discussion by RobsonKing** [sidekiq/sidekiq#5563] [Q&A]
I know that I can get the json statistics for sidekiq from /sidekiq/stats

However I want to provide my own statistics... How many of a certain type of job are running etc.

I know how to find the values that I want to return.  What I am struggling with is **how to return them**?

I could add my own controller to the rails app (controllers/my_sidekiq_controler) and return json from a custom route.  But then I have to deal with authentication etc.  

I see that there is a way to extend the sidekiq web ui with custom tabs (Sidekiq::Web.register MyAddition).  But all the example that I see return something rendered by ERB (because they are meant to run in a tab on the web ui). 

**Is there a way to resister something that just returns json?** 

or is there some other way that i can share the same sidekiq auth structure that the web ui uses?

**Comment by mperham** [comment]
You could monkeypatch anything in your initializer, which could add a new route to Sidekiq::Web. That’s probably easiest.
